### ISSUE TITLE
Support for PulseAudio sources in the sound block

### ISSUE THREAD

**Issue by fedario** [greshake/i3status-rust#277]
Hi there,

I'm trying go set up two sound blocks, one for input and one for output.
The wiki states, that the default for the `name` field is `@DEFAULT_SINK@`, so I thought using `@DEFAULT_SOURCE@` would work for the sound input block. 
However, setting `name` explicitly to either `@DEFAULT_SINK@` or `@DEFAULT_SOURCE@` results in the block showing 00% volume and trying to change it results in i3status-rust crashing with "Error in block 'Sound': volume unknown"

My sound block config:
```
[[block]]
block = "sound"
driver = "auto"
name = "@DEFAULT_SINK@"
on_click = "pactl set-sink-mute @DEFAULT_SINK@ toggle"
```

Btw: I think this `on_click` behavior could be a reasonable default, what do you think?

**Comment by atheriel** [comment]
As to your second point: that's already the default right click behaviour at the moment. I'm not sure about your actual issue, though -- do other `pactl` commands work correctly?

**Comment by fedario** [comment]
Oh, I didn't know right click did anything in blocks.
Yea, `pactl` itself works. But setting `name = "@DEFAULT_SOURCE@"` or  ` name = "@DEFAULT_SINK@"` crashes the whole status bar.

**Comment by atheriel** [comment]
cc @SWW13

**Comment by SWW13** [comment]
PulseAudio distinguishes between source (input) and sink (output) and has a different functions for each.

The current implementation only handles sinks, supporting sources requires a bit of rework. So this is basically a feature request.

The crash is the result of the missing volume, I'm not sure if we should throw an error or not.

**Comment by fedario** [comment]
Thank you for clarifying.
I'm going to check the code of the sound block and see if my Rust skills are good enough to implement the source feature.

**Comment by ammgws** [comment]
Resolved by #740
